### PR TITLE
Add gemfile variables

### PR DESCRIPTION
**Pull request by joe-sharp**
Raised in https://github.com/thoughtbot/appraisal/pull/191#issuecomment-1248935068, it seems desirable to be able to use variables in `customize_gemfiles` so that custom instructions can be provided. I have added the most likely used options, but I am open to additional ideas.

I also wanted to rewrite the specs to make them a bit easier to troubleshoot if they fail. I added a spec to cover the issue fixed in #222 and coverage for the new variables feature added here.

### CODE REVIEW

**Comment by joe-sharp** [comment]
This PR contains the commit from https://github.com/thoughtbot/appraisal/pull/222 but is otherwise ready for review

**Comment by nickcharlton** [comment]
Nice! Thanks for doing this. Could you add a line to the README mentioning it?

**Comment by joe-sharp** [comment]
@nickcharlton , done and I added a few more useful variables as well. All documented and covered by specs! 😁

**Review by nickcharlton — COMMENTED** [review]
Nice, thanks for doing this!

File: README.md [hunk 1]
```diff
@@ -169,8 +178,8 @@ customize_gemfiles do
     heading: <<~HEADING
       frozen_string_literal: true
 
-      File has been generated by Appraisal, do NOT modify it directly!
-      See the conventions at https://example.com/
+      `%{gemfile}` has been generated by Appraisal, do NOT modify it or `%{lockfile}` directly!
+      Make the changes to the "%{appraisal}" block in `Appraisals` instead. See the conventions at https://example.com/
```
[the remark this answers is not in the document]
Comment (joe-sharp) [reply README.md:182]: 🤦🏼‍♂️  Thanks for catching that, thought I had added that to the commit before I pushed!

**Comment by joe-sharp** [comment]
@nickcharlton I believe this is ready to merge. Thanks!

**Review by nickcharlton — APPROVED** [review]

**Pull request merged**